core/src/mast/mod.rs

```diff
@@ -458,6 +458,23 @@ impl MastForest {
             .expect("MAST forest contains more than 2^32 procedures.")
     }
 
+    /// Returns the [Word] representing the hash of the procedures in this forest.
+    pub fn digest(&self) -> Word {
+        self.digest_from_nodes(&self.roots)
+    }
+
+    /// Returns the [Word] representing the content hash of a subset of [`MastNodeId`]s.
+    pub fn digest_from_nodes<'a>(
+        &self,
+        node_ids: impl IntoIterator<Item = &'a MastNodeId>,
+    ) -> Word {
+        let digests = BTreeSet::from_iter(node_ids.into_iter().map(|&id| self[id].digest()));
+        digests
+            .into_iter()
+            .reduce(|a, b| crate::crypto::hash::Rpo256::merge(&[a, b]))
+            .unwrap()
```

Comment:
This implementation runs the risk of creating a different hash depending on the order the `node_ids` are passed. I *assume* that this is mitigated in `BTreeSet`, but I can't find any such guarantees in the docs, so we shouldn't assume it. One option would be:

- Create a `Vec` from `node_ids` (or probably change interface to accept a `Vec`)
- sort `Vec`
- call `Vec::dedup()`
- run your `reduce(merge())` on that `Vec`

Reply:
Ah! Good to know.